src/solvers/README.md

```diff
@@ -62,7 +62,111 @@ different decision procedures, roughly one per directory.
 
 \section flattening-section Flattening
 
-To be documented.
+**Key classes:**
+* \ref propt
+* \ref boolbvt
+* \ref bv_utilst
+
+The main class in this folder is \ref boolbvt which wraps a variety of 
+helper methods and classes (including inherited ones) and acts as an interface 
+for transforming \ref exprt into boolean formula and then solving it.
+
+Many of its methods are focused around transforming a particular \ref exprt 
+into a vector of \ref literalt and then passing them to a \ref propt 
+for formula building / solving.
+
+The primary methods are:
+
+`bvt boolbvt::convert_bitvector(const exprt &expr)`
+
+Which takes an exprt then calls the associated transformation functions to
+generate the \ref literalt vector to then pass to the internal \ref propt instance.
+
+`const bvt & boolbvt::convert_bv(const exprt &expr, optionalt<std::size_t> expected_width)`
+
+Similar to convert_bitvector except it also provides basic caching and
+freezing results for incremental generation. It calls convert_bitvector 
+internally.
+
+`literalt boolbvt::convert_rest(const exprt &expr)`
+
+(Note: I'm not sure why this is split from the normal convert_bitvector, 
+but it's probably worth mentioning)
+
+`void post_process()`
+
+Performs any post-processing, which is normally adding constraints.
+
+Some key classes:
+
+\ref propt is instance of whichever solver is currently being used. This
+inherits from `decision_proceduret` whose interface has a fuller explanation 
+in the "General Interfaces" subsection.
```

Comment:
It's nice to have an explanation here, but it would be even better to have it directly in `propt`. :factory:

Reply:
The comments are (mostly) focused on how they're used in relation to the main class rather than overall descriptions. Also because I didn't think I'd be able to do the classes justice in really explaining their intention in an isolated sense.